modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/ImageMosaicReader.java

```diff
@@ -470,6 +470,8 @@ public ImageMosaicReader(Object source, Hints uHints) throws IOException {
                 } 
                 initReaderFromURL(source, localHints);
             } catch (Exception e) {
+                LOGGER.log(Level.FINE, "Exception generated while trying to initialize ImageMosaicReader: "
+                        + e.getMessage(), e);
                 throw new DataSourceException(e);
```

Comment:
Just wrapping for an IOException? This is a chance to provide a message and caused by - so that the caller can get more details.